### PR TITLE
Salto 2503 skip fetching of disabled features

### PR DESCRIPTION
**Pull request by Shulik95**
_Disabled features will be skipped during fetch instead of failing the entire fetch_

---

_None_

---
__Release Notes__: 
_Netsuite Adapter_:
* Trying to fetch with disabled features will not fail the entire fetch but will skip the features.

---
_User Notifications_: 
_Disabled features will be skipped during fetch instead of failing the entire fetch_

### CODE REVIEW

File: packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts [hunk 1]
```diff
@@ -597,15 +597,18 @@ export default class SoapClient {
...
       SEARCH_RESPONSE_SCHEMA,
       response
     )) {
       log.error(`Got invalid response from search request with SOAP api of type ${type}. Errors: ${this.ajv.errorsText()}. Response: ${JSON.stringify(response, undefined, 2)}`)
       throw new Error(`VALIDATION_ERROR - Got invalid response from search request of type ${type}. Errors: ${this.ajv.errorsText()}. Response: ${JSON.stringify(response, undefined, 2)}`)
     }
-    log.debug(`Finished sending search request for page 1/${Math.max(response.searchResult.totalPages, 1)} of type ${type}`)
-    return response
+    if ('recordList' in response.searchResult) {
+      log.debug(`Finished sending search request for page 1/${Math.max(response.searchResult.totalPages, 1)} of type ${type}`)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts:608]:
```suggestion
      log.debug(`Finished sending search request for page 1/${response.searchResult.totalPages} of type ${type}`)
```

File: packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts [hunk 2]
```diff
@@ -597,15 +597,18 @@ export default class SoapClient {
...
+    if (!this.ajv.validate<SearchResponse | SearchErrorResponse>(
       SEARCH_RESPONSE_SCHEMA,
       response
     )) {
       log.error(`Got invalid response from search request with SOAP api of type ${type}. Errors: ${this.ajv.errorsText()}. Response: ${JSON.stringify(response, undefined, 2)}`)
       throw new Error(`VALIDATION_ERROR - Got invalid response from search request of type ${type}. Errors: ${this.ajv.errorsText()}. Response: ${JSON.stringify(response, undefined, 2)}`)
     }
-    log.debug(`Finished sending search request for page 1/${Math.max(response.searchResult.totalPages, 1)} of type ${type}`)
-    return response
+    if ('recordList' in response.searchResult) {
```
Comment (omrilit) [inline packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts:607]: Is it safe to count on `recordList`? maybe better to use `totalPages`?
Not sure about it, just felt more intuitive to me.

File: packages/netsuite-adapter/test/client/suiteapp_client.test.ts [hunk 3]
```diff
@@ -611,7 +611,6 @@ describe('SuiteAppClient', () => {
             ],
           },
         })
-
```
Comment (omrilit) [inline packages/netsuite-adapter/test/client/suiteapp_client.test.ts:614]: nit: redundant

**Review by omrilit — COMMENTED** [review]
Looks good, had only small comments

File: packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts (hunk 2 above)
Comment (Shulik95) [reply packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts:607]: Changed it to check `totalPages`. But out of curiosity - why is `totalPages` better to use over `recordList`? 
Looking at the schema, both fields are required int it so I don't see any difference in this case...

Comment (omrilit) [reply packages/netsuite-adapter/src/client/suiteapp_client/soap_client/soap_client.ts]: As I said "just felt more intuitive to me" 🤷

**Review by omrilit — APPROVED** [review]
LGTM!

**Pull request merged**